Realm/Realm.UnityWeaver/UnityWeaver.cs

```diff
@@ -28,17 +28,40 @@
 using UnityEditor.Build;
 using UnityEditor.Build.Reporting;
 using UnityEditor.Compilation;
+using UnityEngine;
 
 namespace RealmWeaver
 {
     // Heavily influenced by https://github.com/ExtendRealityLtd/Malimbe and https://github.com/fody/fody
     public class UnityWeaver : IPostBuildPlayerScriptDLLs
     {
+        private const string EnableAnalyticsPref = "realm_enable_analytics";
+        private const string EnableAnalyticsMenuItemPath = "Realm/Enable build-time analytics";
+
+        private static bool _analyticsEnabled;
+
+        private static bool AnalyticsEnabled
+        {
+            get => _analyticsEnabled;
+            set
+            {
+                _analyticsEnabled = value;
+                EditorPrefs.SetBool(EnableAnalyticsPref, value);
+                Menu.SetChecked(EnableAnalyticsMenuItemPath, value);
+            }
+        }
+
         public int callbackOrder => 0;
 
         [InitializeOnLoadMethod]
         public static void Initialize()
         {
+            // We need to call that again after the editor is initialized to ensure that we populate the checkmark correctly.
```

Comment:
Interesting quirk

Reply:
Yeah, apparently InitializeOnLoad is invoked before the editor has fully rendered, so updating UI elements (like adding a checkmark to the menu item) won't do anything.